version: "3.9"

services:
  postgres:
    image: ${POSTGIS_CONTAINER_IMAGE:-postgis/postgis}:${TEST_TAG:-latest}
    volumes:
      - 'postgres-data:/var/lib/postgresql/data'
    networks:
      - "default"
    ports:
    - '5432:5432'
    environment:
      - 'POSTGRES_DB=${POSTGRES_DB:-peaks}'
      - 'POSTGRES_USER=${POSTGRES_USER:-mfi}'
      - 'POSTGRES_PASSWORD=${POSTGRES_PASSWORD:-peaks#local}'
  api:
    build:
      context: ./api
      dockerfile: Dockerfile
      args:
        PYTHON_IMAGE: ${PYTHON_IMAGE:-python:3.10-slim}
    image: ${API_CONTAINER_IMAGE:-m-peaks-mfi/api}:${TEST_TAG:-latest}
    networks:
      - "default"
    ports:
      - '5001:5001'
    environment:
      - VERSION=${VERSION:-1.0.0}
      - HTTP_HOST=0.0.0.0
      - HTTP_PORT=5001
      - INSECURE_CORS=1
      - API_LOG_LEVEL=${API_LOG_LEVEL:-INFO}
      - DB_HOST=postgres
      - DB_NAME=peaks
      - DB_USER=mfi
      - DB_PASSWORD=${POSTGRES_PASSWORD:-peaks#local}

volumes:
  postgres-data:
    driver: "local"
 
networks:
  default:
    driver: bridge
